mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -438,7 +446,10 @@ def handle_settings_change(self, message):
             if remote_settings is not None:
                 LOG.info('Updating settings for skill ' + self.name)
                 self.settings.update(**remote_settings)
-                save_settings(self.settings_write_path, self.settings)
+                if isinstance(self.settings, JsonStorage):
+                    self.settings.store()  # else not initialized and it's a dict
```

Comment:
Changing settings object type might be worth its own discussion/PR (https://github.com/MycroftAI/mycroft-core/issues/2912)

Reply:
this change quickly grew in scope, will split into its own PR